File: rb/features.rb
# PlaystationStore SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module PlaystationStoreFeatures
  def self.make_feature(name)
    case name
    when "base"
      PlaystationStoreBaseFeature.new
    when "ratelimit"
      PlaystationStoreRatelimitFeature.new
    when "retry"
      PlaystationStoreRetryFeature.new
    when "test"
      PlaystationStoreTestFeature.new
    when "timeout"
      PlaystationStoreTimeoutFeature.new
    else
      PlaystationStoreBaseFeature.new
    end
  end
end

File: py/features.py
# PlaystationStore SDK feature factory

from feature.base_feature import PlaystationStoreBaseFeature
from feature.test_feature import PlaystationStoreTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PlaystationStoreBaseFeature(),
        "test": lambda: PlaystationStoreTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
